### PR TITLE
docs(repo): discourage change-detector tests

### PR DESCRIPTION
**Pull request by mdrxy**
Clarify the repository-wide testing guidance so tests verify observable behavior instead of mirroring implementation details or incidental interactions. The guidance now asks contributors to rewrite or remove change-detector tests and links to the Google Testing Blog article for context.

Made by [Open SWE](https://openswe.vercel.app/agents/5812fc60-da61-50b7-8cbf-b28a9b6296f1)

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ Every feature or bugfix needs unit coverage.
 
 - Put network-free tests in `tests/unit_tests/` and networked tests in `tests/integration_tests/`.
 - Do not add `@pytest.mark.asyncio`; packages use `asyncio_mode = "auto"`.
-- Test behavior rather than duplicating implementation logic. Cover edge cases and keep tests deterministic.
+- Test observable behavior rather than duplicating implementation logic. Do not add change-detector tests that merely restate the current code structure or assert incidental interactions, such as internal call order, without proving meaningful behavior. Refactors that preserve behavior should not require mechanical test updates; rewrite or remove tests that do. Cover edge cases, keep tests deterministic, and see [Change-Detector Tests Considered Harmful](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html).
```
Comment (mdrxy) [inline AGENTS.md]:
```suggestion
- Test observable behavior rather than duplicating implementation logic. Do not add change-detector tests that merely restate the current code structure or assert incidental interactions, such as internal call order, without proving meaningful behavior. Refactors that preserve behavior should not require mechanical test updates; rewrite or remove tests that do. Cover edge cases and keep tests deterministic.
```
